### ISSUE TITLE
Support for malformed unstructured fields containing encoded words

### ISSUE THREAD

**Issue by BryanLeong** [stalwartlabs/mail-parser#29]
I've come across a number of emails where the subject, which contains encoded words, was modified by the recipients' mail server such that the final subject became something like:

```
[SUSPECTED SPAM]=?utf-8?B?VGhpcyBpcyB0aGUgb3JpZ2luYWwgc3ViamVjdA==?=
```

I understand this does not get decoded as it is missing a space before the encoded word as required in [the spec](https://datatracker.ietf.org/doc/html/rfc2047#section-5)
> Ordinary ASCII text and 'encoded-word's may appear together in the
    same header field.  However, an 'encoded-word' that appears in a
    header field defined as '*text' MUST be separated from any adjacent
    'encoded-word' or 'text' by 'linear-white-space'.

Would it be possible to add support for parsing these types of malformed fields, seeing as mail servers which do this are relatively common?

**Comment by mdecimus** [comment]
Sure, the library already handles different types of malformed messages so it's a good idea to support broken encoded headers. Do you have other samples of broken encoded headers? Or is it mostly missing spaces between the prefix and the beginning of the encoded text?

**Comment by BryanLeong** [comment]
> missing spaces between the prefix and the beginning of the encoded text

Mostly ☝🏼, but also headers with empty encoded words, eg: `Some text here =?utf-8?Q??=`

**Comment by mdecimus** [comment]
Thanks, I'll look into it shortly.

**Comment by BryanLeong** [comment]
Thanks! Confirmed that it fixes the issues mentioned above.

I found a couple more broken ones though, this time they seem to be caused by newlines in the middle of the encoded words 😅 

```
Subject: =?utf-8?Q?Hello
 _there!?=
```
Expected: `Hello there!`
Actual: `=?utf-8?Q?Hello _there!?=`

**Comment by mdecimus** [comment]
Done! I just pushed the fix which also supports multi-line base64 encoded words.

**Comment by BryanLeong** [comment]
Small issue: the folding whitespace gets included in the output instead of being removed together with the newline

**Comment by mdecimus** [comment]
Just fixed it.

**Comment by BryanLeong** [comment]
All good now. Thanks! 🙏🏼

**Comment by mdecimus** [comment]
Perfect, just published version 0.6.1 to crates.io.
